app/src/main/java/net/gsantner/markor/activity/BackupRestoreHelper.java

```diff
@@ -0,0 +1,185 @@
+package net.gsantner.markor.activity;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.support.v4.app.FragmentManager;
+import android.widget.Toast;
+
+import net.gsantner.markor.R;
+import net.gsantner.markor.ui.FilesystemViewerCreator;
+import net.gsantner.markor.util.AppSettings;
+import net.gsantner.opoc.ui.FilesystemViewerData;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class BackupRestoreHelper {
+
+    private static final String SAVE_NAME = "markor_settings_backup%s.zip";
+    private static final int BUFFER_SIZE = 2048;
+
+    private static String[] BACKUP_PATHS = {
+            "shared_prefs/action_order.xml",
+            "shared_prefs/app.xml",
+            "shared_prefs/datetime_dialog_settings.xml",
+            "shared_prefs/search_replace_dialog_settings.xml",
```

Comment:
suggestiong whole folder, and excluding. instead of whitelist

Reply:
I thought of this, but this causes some additional complexity.

When restoring the backup we will have to have a whitelist anyway, otherwise we will have to iterate over the contents of the zip file and guess whether a file should be restored or not.

Reply:
@gsantner Have found a good fix - using both a blacklist and a whitelist. This works well and is now ready for review / merge